internals/garbage-collector.rst

```diff
@@ -356,37 +356,70 @@ follows these steps in order:
    the reference counts fall to 0, triggering the destruction of all unreachable
    objects.
 
-Optimization: generations
-=========================
+Optimization: incremental collection
+====================================
 
-In order to limit the time each garbage collection takes, the GC
-implementation for the default build uses a popular optimization:
-generations. The main idea behind this concept is the assumption that most
-objects have a very short lifespan and can thus be collected soon after their
-creation. This has proven to be very close to the reality of many Python
+In order to limit the time each garbage collection takes, the GC implementation
+for the default build uses incremental collection with two generations.
+
+Generational garbage collection takes advantage of what is known as the weak
+generational hypothesis: Most objects die young.
+This has proven to be very close to the reality of many Python
 programs as many temporary objects are created and destroyed very quickly.
 
 To take advantage of this fact, all container objects are segregated into
-three spaces/generations. Every new
-object starts in the first generation (generation 0). The previous algorithm is
-executed only over the objects of a particular generation and if an object
-survives a collection of its generation it will be moved to the next one
-(generation 1), where it will be surveyed for collection less often. If
-the same object survives another GC round in this new generation (generation 1)
-it will be moved to the last generation (generation 2) where it will be
-surveyed the least often.
-
-The GC implementation for the free-threaded build does not use multiple
-generations.  Every collection operates on the entire heap.
+two generations: young and old. Every new object starts in the young generation.
+
+To detect and collect all unreachable cycles in the heap, the garbage collector
+must scan the whole heap. This whole heap scan is called a full scavenge.
+
+To limit the time each garbage collection takes, the detection and collection
+algorithm is executed only on a portion of the heap called an increment.
+For each full scavenge, the increments will cover the whole heap.
+
+Each increment, the portion of the heap scanned by a single collection is made up
+of three parts:
+
+* The young generation
+* The least recently scanned fraction of the old generation.
+* All objects reachable from those objects that have not yet been scanned this cycle.
+
+Any objects surviving this collection are moved to the old generation.
+The old generation is composed of two lists, scanned and unscanned.
+(The implementation refers to the unscanned part as ``pending`` and the scanned part
+as ``visited``).
+Survivors are moved to the back of the scanned list. The old part of increment is taken
+from the front of the unscanned list.
+
+When a cycle starts, no objects in the heap are considered to have been scanned.
```

Comment:
Using cycle of a "reference cycle" and a "gc execution" it's a bit confusing, can we use other terminology?

Reply:
I've used "full scavenge"